### PR TITLE
Implement SaplingGrowTreeEvent and BlockEvent.FarmlandTrampleEvent

### PR DESCRIPTION
**Pull request by famous1622**
Both of these events are required for CoFH Core. Doing them in the same PR as they're both relatively simple.

### CODE REVIEW

File: patchwork-events-world/src/main/java/net/minecraftforge/event/world/SaplingGrowTreeEvent.java [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+ * This event is fired during sapling growth in
+ * {@link net.minecraft.block.SaplingBlock#generate(IWorld, BlockPos, BlockState, Random)}.<br>
+ * <br>
+ * {@link #pos} contains the coordinates of the growing sapling. <br>
+ * {@link #rand} contains an instance of Random for use. <br>
+ * <br>
+ * This event is not cancellable.<br>
+ * <br>
+ * This event has a result.<br>
+ * This result determines if the sapling is allowed to grow. <br>
```
Comment (kitlith) [inline patchwork-events-world/src/main/java/net/minecraftforge/event/world/SaplingGrowTreeEvent.java:39]: IIRC generally we turn Forge's use of linebreaks into proper paragraphs.

File: patchwork-events-world/src/main/java/net/minecraftforge/event/world/BlockEvent.java [hunk 2]
```diff
@@ -109,4 +110,27 @@ public boolean isCancelable() {
 			return true;
 		}
 	}
+
+	/**
+	 * Fired when when farmland gets trampled
+	 * This event is cancellable.
+	 */
+	public static class FarmlandTrampleEvent extends BlockEvent {
```
Comment (TheGlitch76) [inline patchwork-events-world/src/main/java/net/minecraftforge/event/world/BlockEvent.java:118]: Cancellable in Forge, but not here

File: patchwork-events-world/src/main/java/net/minecraftforge/event/world/SaplingGrowTreeEvent.java (hunk 1 above)
Comment (TheGlitch76) [reply patchwork-events-world/src/main/java/net/minecraftforge/event/world/SaplingGrowTreeEvent.java:39]: Kit is correct.

File: patchwork-events-world/src/main/java/net/patchworkmc/impl/event/world/WorldEvents.java [hunk 3]
```diff
@@ -57,4 +64,19 @@ public static void onWorldUnload(IWorld world) {
 	public static void onWorldSave(IWorld world) {
 		MinecraftForge.EVENT_BUS.post(new WorldEvent.Save(world));
 	}
+
+	public static boolean onFarmlandTrample(World world, BlockPos pos, BlockState state, float fallDistance, Entity entity) {
+		//TODO: In forge, the possibility of trampling is handled by IForgeEntity.canTrample
```
Comment (TheGlitch76) [inline patchwork-events-world/src/main/java/net/patchworkmc/impl/event/world/WorldEvents.java:69]:
```suggestion
		// TODO: In forge, the possibility of trampling is handled by IForgeEntity.canTrample
```

File: patchwork-events-world/src/main/java/net/minecraftforge/event/world/SaplingGrowTreeEvent.java [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+ * <p>{@link #pos} contains the coordinates of the growing sapling.
+ * {@link #rand} contains an instance of Random for use.
+ *
+ * <p>This event is not cancellable.
+ *
+ * <p>This event has a result.
+ * This result determines if the sapling is allowed to grow.
+ */
+
+public class SaplingGrowTreeEvent extends WorldEvent {
```
Comment (TheGlitch76) [inline patchwork-events-world/src/main/java/net/minecraftforge/event/world/SaplingGrowTreeEvent.java:43]: This event is marked @HasResult but doesn't override the method here

**Comment by TheGlitch76** [comment]
Please resolve merge conflicts

**Comment by TheGlitch76** [comment]
Failing checkstyle

**Comment by TheGlitch76** [comment]
Lots and lots of merge conflicts and a failing checkstyle

**Review by TheGlitch76 — APPROVED** [review]
Sorry for the wait, lgtm

**Pull request merged**